infrastructure/oss/scanner.go

```diff
@@ -72,11 +74,16 @@ var supportedFiles = map[string]bool{
 	"mix.lock":                true,
 }
 
+// a counter for scans, used for logging
+var scanCount = 1
+
 type Scanner struct {
 	instrumentor  performance.Instrumentor
 	errorReporter error_reporting.ErrorReporter
 	analytics     ux2.Analytics
 	cli           cli.Executor
+	mutex         *sync.Mutex
+	runningScans  map[string]*scans.ScanProgress
```

Comment:
Would it make sense to use an AtomicMap here to automatically synchronize access?

Reply:
I don't think so

We're only accessing this map inside locked sections, and there are more actions being done inside these locked sections that are unrelated to the map (for example checking the `ScanProgress` value